partiql-conformance-test-generator/src/generator.rs

```diff
@@ -42,24 +41,23 @@ fn namespace_to_module(namespace: &Namespace) -> Module {
 fn test_case_to_function(test_case: &TestCase) -> Function {
     let mut test_fn: Function = Function::new(&test_case.test_name);
     test_fn.attr("test");
-    match &test_case.test_kind {
-        Parse(ParseTestCase { parse_assertions }) => {
-            test_fn.line(format!("let query = r#\"{}\"#;", &test_case.statement));
-            test_fn.line("let res = partiql_parser::Parser::default().parse(query);");
-            match parse_assertions {
-                ParseAssertions::ParsePass => test_fn
-                    .line(r#"assert!(res.is_ok(), "For `{}`, expected `Ok(_)`, but was `{:#?}`", query, res);"#),
-                ParseAssertions::ParseFail => test_fn
-                    .line(r#"assert!(res.is_err(), "For `{}`, expected `Err(_)`, but was `{:#?}`", query, res);"#),
-            };
+    test_fn.line(format!("let statement = r#\"{}\"#;", &test_case.statement));
+    for assertion in &test_case.assertions {
+        match assertion {
+            Assertion::SyntaxSuccess => {
+                test_fn.line("let res = partiql_parser::Parser::default().parse(statement);");
+                test_fn.line(r#"assert!(res.is_ok(), "For `{}`, expected `Ok(_)`, but was `{:#?}`", statement, res);"#);
+            }
+            Assertion::SyntaxFail => {
+                test_fn.line("let res = partiql_parser::Parser::default().parse(statement);");
+                test_fn.line(r#"assert!(res.is_err(), "For `{}`, expected `Err(_)`, but was `{:#?}`", statement, res);"#);
+            }
+            Assertion::NotYetImplemented => {
+                // for `NotYetImplemented` assertions, add the 'ignore' annotation to the test case
+                test_fn.attr("ignore");
```

Comment:
The [.attr](https://docs.rs/codegen/latest/codegen/struct.Function.html#method.attr) from the code generation crate adds the `[#ignore]` attribute to the test function, so I don't think we can add the additional `- not yet implemented`. Otherwise, the generated code

```
#[test]
#[ignore - not yet implemented]
fn some_ignored_test() {
    ...
}
```
results in a compilation error:
```
2 | #[ignore - not yet implemented]
  |          ^ expected one of `(`, `::`, `=`, `[`, `]`, or `{`

error: expected one of `(`, `::`, `=`, `[`, `]`, or `{`, found `-`
```

Reply:
Oh cool. I didn't know about the [MetaNameValueStr](https://doc.rust-lang.org/reference/attributes.html#meta-item-attribute-syntax) syntax to give a reason why the test is ignored. Will update to use.